tests/parser/parser.cpp

```diff
@@ -1874,3 +1854,86 @@ TEST(Parser, xmlComments)
+{
+    const std::string input =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component name=\"component\">\n"
+        "    <variable name=\"A\" initial_value=\"1.0\"/>\n"
+        "    <variable name=\"B\" initial_value=\"-1.0\"/>\n"
+        "    <math xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "      <apply>\n"
+        "        <eq/>\n"
+        "        <ci>C</ci>\n"
+        "        <apply>\n"
+        "          <plus/>\n"
+        "          <ci>A</ci>\n"
+        "          <ci>B</ci>\n"
+        "        </apply>\n"
+        "      </apply>\n"
+        "    </math>\n"
+        "  </component>\n"
+        "</model>\n";
+
+    libcellml::Parser parser;
+    libcellml::Printer printer;
+
+    libcellml::ModelPtr model1 = parser.parseModel(input);
+    std::string output1 = printer.printModel(model1);
+
+    libcellml::ModelPtr model2 = parser.parseModel(output1);
+    std::string output2 = printer.printModel(model2);
+
+    libcellml::ModelPtr model3 = parser.parseModel(output2);
+    std::string output3 = printer.printModel(model3);
+
+    EXPECT_EQ(input, output3);
+}
+
+TEST(Parser, repeatedMathParsePrintBehaviourWithReset)
+{
+    const std::string input =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "  <component name=\"component\">\n"
+        "    <variable name=\"variable1\" units=\"blob\" id=\"v1id\"/>\n"
+        "    <variable name=\"variable2\" units=\"blob\" id=\"v2id\"/>\n"
+        "    <reset variable=\"variable1\" test_variable=\"variable2\" order=\"1\" id=\"r1id\">\n"
+        "      <test_value>\n"
+        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\" id=\"math1when1\">\n"
+        "          <apply>\n"
+        "            <eq/>\n"
+        "            <ci>variable1</ci>\n"
+        "            <cn>3.4</cn>\n"
+        "          </apply>\n"
+        "        </math>\n"
+        "      </test_value>\n"
+        "      <reset_value>\n"
+        "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\" id=\"math2when1\">\n"
+        "          <apply>\n"
+        "            <eq/>\n"
+        "            <ci>variable1</ci>\n"
+        "            <cn>9.0</cn>\n"
+        "          </apply>\n"
+        "        </math>\n"
+        "      </reset_value>\n"
+        "    </reset>\n"
+        "  </component>\n"
+        "</model>\n";
+
+    libcellml::Parser parser;
+    libcellml::Printer printer;
+
+    libcellml::ModelPtr model1 = parser.parseModel(input);
+    std::string output1 = printer.printModel(model1);
+
+    libcellml::ModelPtr model2 = parser.parseModel(output1);
+    std::string output2 = printer.printModel(model2);
+
+    libcellml::ModelPtr model3 = parser.parseModel(output2);
+    std::string output3 = printer.printModel(model3);
+
+    EXPECT_EQ(input, output3);
```

Comment:
Same as above, i.e. no real need for `mdoel3` and `output3`. The following ought to be enough:

```
EXPECT_EQ(input, output2);
```